java-frontend/src/test/java/org/sonar/java/JavaSquidTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java;
+
+import com.google.common.collect.Lists;
+import com.google.common.io.Files;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.internal.DefaultInputFile;
+import org.sonar.api.batch.rule.CheckFactory;
+import org.sonar.api.batch.sensor.highlighting.TypeOfText;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+import org.sonar.api.internal.SonarRuntimeImpl;
+import org.sonar.api.issue.NoSonarFilter;
+import org.sonar.api.measures.FileLinesContext;
+import org.sonar.api.measures.FileLinesContextFactory;
+import org.sonar.api.utils.Version;
+import org.sonar.plugins.java.api.JavaResourceLocator;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class JavaSquidTest {
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+
+  @Test
+  public void number_of_visitors_in_sonarLint_context_LTS() throws Exception {
+    testVersion(Version.create(5, 6));
+  }
+  @Test
+  public void number_of_visitors_in_sonarLint_context_60() throws Exception {
+    testVersion(Version.create(6, 0));
+  }
+
+  private void testVersion(Version version) throws IOException {
+    SensorContextTester context = SensorContextTester.create(temp.getRoot());
+
+    // set up a file to analyze
+    File file = temp.newFile();
+    Files.write(Files.toString(new File("src/test/java/org/sonar/java/JavaSquidTest.java"), StandardCharsets.UTF_8), file, StandardCharsets.UTF_8);
```

Comment:
I would prefer to use a dedicated test file rather than having the test class testing itself. Any change to the file (spacing, slight refactor, adding new tests, changing imports, ...) will have consequences on results of some of the assertions.